### ISSUE TITLE
`migrate-log.py --convert` exits 1 with a traceback after a fully successful conversion, whenever an observation title contains a character the console encoding cannot represent

### ISSUE THREAD

**Issue by YeQiuApex** [rebelytics/one-skill-to-rule-them-all#78]
Migrating a 59-entry v2 log on Windows 11 (Python 3.12.10, French locale, console
codepage cp1252), `scripts/migrate-log.py --convert` wrote all 59 files correctly and
then died:

```
  File "scripts/migrate-log.py", line 459, in main
    print(f"  #{r['id']:<5} {why:<42} {r['title'][:56]}")
UnicodeEncodeError: 'charmap' codec can't encode character '→' in position 70
```

Exit code 1. Files on disk: 59/59, complete and correct.

The `→` is not a literal in the script. It comes from `r['title']` — one of my own
observations is titled *"Chercher un pont A→B sans chercher B→A fait rater la moitié de
la réponse"*. So the trigger is log content, not the source: any migration whose log
contains a title with an em-dash, an arrow, a curly quote, a CJK character or an emoji
hits this on a default Windows console. That is not an exotic case for a tool whose
whole purpose is migrating prose written by agents.

**Why this is worse than a cosmetic crash.** Reading `main()`, every write completes
before any of the reporting runs: the 59 files are written at lines 440-443, `.id-floor`
at 447-449, and the first `print` of the report is line 451. The failure is therefore
located strictly after the entire state change, in the summary — but nothing on the
operator's side says so. A traceback plus exit 1 on a one-shot, state-changing,
upgrade-only script reads as "the migration failed". The natural responses — roll back,
re-run, restore the backup — are all wrong, and one of them (re-running against a log
that has already been renamed per step 7) will produce a second, more confusing failure.

**The check-only rehearsal cannot catch it.** `references/migration.md` step 3 has the
operator run `--check` over every log they have, precisely so the conversion is
rehearsed before it touches anything. `--check` returns at line 436, before the block
that prints per-entry titles, so it prints only aggregate counts and exits 0 — verified
on the same log that crashes the convert. The documented safety pass passes cleanly and
then the real run dies. The rehearsal is structurally blind to the failure it exists to
prevent.

**It also destroys the artefact the next step depends on.** Step 6 asks the operator to
compare the report's file count against `grep -c '^### Observation'`. The listing of
flagged entries — the thing step 4's overrides file is written from — is exactly what
never gets printed. So the operator loses the verification input at the same moment they
are handed a false failure signal.

**Reproduction:** any pre-3.0 `log.md` with one non-cp1252 character in an
`### Observation N:` header, on a Windows console:

```
python scripts/migrate-log.py --convert log.md --out observation-log
echo %errorlevel%   ->  1, with all files written
```

`PYTHONIOENCODING=utf-8 python scripts/migrate-log.py --convert ...` succeeds and prints
the full report. That is the workaround, but it has to be known in advance, and nothing
in `migration.md` mentions it.

**Suggestion:** make the script's own output encoding-independent rather than asking the
operator to know about `PYTHONIOENCODING`. The script already requires Python 3.8+, so
one line at the top of `main()` covers it:

```python
for s in (sys.stdout, sys.stderr):
    try:
        s.reconfigure(encoding="utf-8", errors="backslashreplace")
    except AttributeError:
        pass
```

`errors="backslashreplace"` is worth preferring over a bare `utf-8`: it also survives a
console that genuinely cannot render the character, degrading one title into escapes
instead of losing the whole report.

Two smaller things in the same area, if you want them in one pass: the report would be
more robust if the write phase and the report phase were separated by a `try/except`
around the reporting, so a summary failure can never mask a completed migration; and
`open(os.path.join(arch, ".id-floor"), "w")` at line 448 omits `encoding=`, which is
harmless today (it writes digits) but inconsistent with the explicit `encoding="utf-8"`
used for the observation files four lines above.

This is the same family as #66 and the platform-portability theme in #77 — POSIX-shaped
assumptions that fail quietly, or in this case loudly but misleadingly, off POSIX.

**Comment by rebelytics** [comment]
Fixed in `main` and shipped in v3.1.0 today — migrate-log.py reconfigures its output streams to replace unencodable characters; verified exit 0 under cp1252 where the previous version exited 1 with your traceback. Thanks for the report.
